org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/AnnotationTest.java

```diff
@@ -10023,6 +10024,67 @@ public void test297() {
 	runner.javacTestOptions = JavacTestOptions.Excuse.EclipseWarningConfiguredAsError;
 	runner.runNegativeTest();
 }
+// https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2176
+public void testGH2167() {
+	Runner runner = new Runner();
+	runner.customOptions = getCompilerOptions();
+	runner.customOptions.put(CompilerOptions.OPTION_SuppressWarnings, CompilerOptions.ENABLED);
+	runner.customOptions.put(CompilerOptions.OPTION_ReportUnhandledWarningToken, CompilerOptions.WARNING);
+	runner.customOptions.put(CompilerOptions.OPTION_SuppressOptionalErrors, CompilerOptions.ENABLED);
+	runner.customOptions.put(CompilerOptions.OPTION_ReportComparingIdentical, CompilerOptions.ERROR);
+	runner.customOptions.put(CompilerOptions.OPTION_ReportComparingWrapper, CompilerOptions.ERROR);
+
+	runner.expectedCompilerLog =
+		"----------\n" +
+		"1. ERROR in A.java (at line 5)\n" +
+		"	return x1 == x2;\n" +
+		"	       ^^^^^^^^\n" +
+		"Comparing primitive wrapper types or Strings using '=='\n" +
+		"----------\n" +
+		"2. ERROR in A.java (at line 10)\n" +
+		"	return x1 == x2;\n" +
+		"	       ^^^^^^^^\n" +
+		"Comparing primitive wrapper types or Strings using '=='\n" +
+		"----------\n" +
+		"3. ERROR in A.java (at line 15)\n" +
+		"	return x1 == x2;\n" +
+		"	       ^^^^^^^^\n" +
+		"Comparing primitive wrapper types or Strings using '=='\n" +
+		"----------\n" +
+		"4. ERROR in A.java (at line 20)\n" +
+		"	return x1 == x2;\n" +
+		"	       ^^^^^^^^\n" +
+		"Comparing primitive wrapper types or Strings using '=='\n" +
+		"----------\n";
+
+	runner.testFiles = new String[] {
```

Comment:
I'd like to see a few more tests:
* mixed comparison primitive and boxed
* at least one other kind of expression, e.g., a method invocation

Reply:
Done.